src/EventSubscriber/SoftDeleteSubscriber.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\EventSubscriber;
+
+use App\Entity\User;
+use App\Repository\ProgrammeRepository;
+use Doctrine\Bundle\DoctrineBundle\EventSubscriber\EventSubscriberInterface;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Event\LifecycleEventArgs;
+use Gedmo\SoftDeleteable\SoftDeleteableListener;
+
+class SoftDeleteSubscriber implements EventSubscriberInterface
```

Comment:
a better naming would mention that it's for the user soft delete, so `UserDeleteSubscriber`